src/librustc_mir/build/block.rs

```diff
@@ -83,4 +82,117 @@ impl<'a,'tcx> Builder<'a,'tcx> {
             block.unit()
         })
     }
+
+    pub fn stmt_expr(&mut self, mut block: BasicBlock, expr: Expr<'tcx>) -> BlockAnd<()> {
```

Comment:
Do you think this is the best place for this function? To me it sounds like it ought to get itself a new module `build::expr::stmt` or something instead.